### PR TITLE
added dockerfile

### PR DESCRIPTION
**Pull request by wxj6000**
- Added dockerfile for compilation and runtime. 
- Removed cutensor in CMakeLists.txt, it is unnecessary
- Stored dispersion correction in mf object.

### CODE REVIEW

File: gpu4pyscf/lib/cupy_helper/CMakeLists.txt [hunk 1]
```diff
@@ -33,38 +33,4 @@ set_target_properties(cupy_helper PROPERTIES
   LIBRARY_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR})
 find_package(CUDAToolkit)
 
-# based on https://github.com/NVIDIA/CUDALibrarySamples/blob/master/cuTENSOR/CMakeLists.txt
```
Comment (sunqm) [inline gpu4pyscf/lib/cupy_helper/CMakeLists.txt:36]: is cutensor not needed?

Comment (wxj6000) [reply gpu4pyscf/lib/cupy_helper/CMakeLists.txt:36]: cutensor is still needed as a library. But the line 'find_package(CUTENSOR)' in CMakeLists.txt is unnecessary.

Comment (sunqm) [reply gpu4pyscf/lib/cupy_helper/CMakeLists.txt:36]: Your PR removes these code

Comment (wxj6000) [reply gpu4pyscf/lib/cupy_helper/CMakeLists.txt:36]: Right, they are unnecessary either based on my test.

**Pull request merged**